utils/imodels-client-test-utils/src/test-context-providers/imodel/TestIModelRetriever.ts

```diff
@@ -112,10 +113,33 @@ export class TestIModelRetriever {
       authorization: this._testAuthorizationProvider.getAdmin1Authorization(),
       iModelId
     };
-    const changesetGroups: ChangesetGroupMetadata[] = await toArray(this._iModelsClient.changesetGroups.getList(getChangesetGroupListParams));
+    const changesetGroups = await toArray(this._iModelsClient.changesetGroups.getList(getChangesetGroupListParams));
     if (changesetGroups.length !== TestIModelCreator.changesetGroups.length)
       throw new TestSetupError(`${changesetGroups.length} is an unexpected changeset group count for reusable test iModel.`);
 
-    return changesetGroups;
+    return changesetGroups.map((csGroup) => {
+      const changesetIndexes = TestIModelCreator.changesetGroups.find((x) => x.description === csGroup.description)?.changesetIndexes;
+      if (!changesetIndexes)
+        throw new TestSetupError("Could not find expected changeset group by description.");
+      return { id: csGroup.id, description: csGroup.description, changesetIndexes };
+    });
+  }
+
+  private async queryAndValidateChangesets(iModelId: string, changesetGroups: ChangesetGroupMetadata[]): Promise<void> {
+    const getChangesetListParams: GetChangesetListParams = {
+      authorization: this._testAuthorizationProvider.getAdmin1Authorization(),
+      iModelId
+    };
+    const changesets = await toArray(this._iModelsClient.changesets.getMinimalList(getChangesetListParams));
+    if (changesets.length !== this._testIModelFileProvider.changesets.length)
+      throw new TestSetupError(`${changesets.length} is an unexpected changeset count for reusable test iModel.`);
+
+    changesetGroups.forEach((csGroup) => {
```

Comment:
nitpick: we do not really use `.forEach` in the releasable code, mostly we use `for (..`, so it would be good to change this for consistency,

`map` is also not needed here since we do not use the results, a simple `for(...` would be sufficient too.

and maybe it would be more readable to iterate over queried changesets and validate them:
```
for (const changeset of changesets) {
  const expectedGroupId = expectedChangesetGroups.find(group => group.changesets.contains(changeset.Index));
  if (changeset.GroupId !== expectedGroupId)
    throw  ....
}

```

Reply:
Ok, updated